pkg/binlog/filename.go

```diff
@@ -30,14 +31,67 @@ var (
 	ErrInvalidBinlogFilename = errors.New("invalid binlog filename")
 )
 
-// VerifyBinlogFilename verify whether is a valid MySQL/MariaDB binlog filename.
-// valid format is `base + '.' + seq`.
-func VerifyBinlogFilename(filename string) error {
+// Filename represents a binlog filename.
+type Filename struct {
+	BaseName string
+	Seq      string
+}
+
+// LessThan checks whether this filename < other filename.
+func (f Filename) LessThan(other Filename) bool {
+	return f.BaseName == other.BaseName && f.Seq < other.Seq
+}
+
+// GreaterThanOrEqualTo checks whether this filename >= other filename.
+func (f Filename) GreaterThanOrEqualTo(other Filename) bool {
+	return f.BaseName == other.BaseName && f.Seq >= other.Seq
+}
+
+// GreaterThan checks whether this filename > other filename.
+func (f Filename) GreaterThan(other Filename) bool {
+	return f.BaseName == other.BaseName && f.Seq > other.Seq
+}
+
+// ParseFilename parses a string representation binlog filename into a `Filename`.
+func ParseFilename(filename string) (Filename, error) {
+	var fn Filename
 	parts := strings.Split(filename, binlogFilenameSep)
 	if len(parts) != 2 {
-		return errors.Annotatef(ErrInvalidBinlogFilename, "filename %s", filename)
+		return fn, errors.Annotatef(ErrInvalidBinlogFilename, "filename %s", filename)
 	} else if n, err := strconv.Atoi(parts[1]); err != nil || n <= 0 {
-		return errors.Annotatef(ErrInvalidBinlogFilename, "filename %s", filename)
+		return fn, errors.Annotatef(ErrInvalidBinlogFilename, "filename %s", filename)
+	}
+	fn.BaseName = parts[0]
+	fn.Seq = parts[1]
+	return fn, nil
+}
+
+// VerifyFilename verifies whether is a valid MySQL/MariaDB binlog filename.
+// valid format is `base + '.' + seq`.
+func VerifyFilename(filename string) error {
+	_, err := ParseFilename(filename)
+	return errors.Trace(err)
+}
+
+// GetFilenameIndex returns a float64 index value (seq number) of the filename.
+func GetFilenameIndex(filename string) (float64, error) {
```

Comment:
why `float64`, not `int64`?

Reply:
changed to `int64` in https://github.com/pingcap/dm/pull/140/commits/d7f60d5dcf1f9caa3fa89c0ad8bbb6821281bd1e.